Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1622,8 +1622,11 @@ void FrameLoader::loadWithDocumentLoader(DocumentLoader* loader, FrameLoadType t
         oldDocumentLoader->setLastCheckedRequest(ResourceRequest());
         policyChecker().stopCheck();
         RELEASE_ASSERT(!isBackForwardLoadType(policyChecker().loadType()) || history().provisionalItem());
-        policyChecker().checkNavigationPolicy(ResourceRequest(loader->request()), ResourceResponse { }  /* redirectResponse */, oldDocumentLoader.get(), WTFMove(formState), [this, protectedFrame = Ref { m_frame }] (const ResourceRequest& request, WeakPtr<FormState>&&, NavigationPolicyDecision navigationPolicyDecision) {
-            continueFragmentScrollAfterNavigationPolicy(request, navigationPolicyDecision == NavigationPolicyDecision::ContinueLoad);
+        RefPtr<SecurityOrigin> requesterOrigin;
+        if (auto& requester = loader->triggeringAction().requester())
+            requesterOrigin = requester->securityOrigin.copyRef();
```

Comment:
Kind of surprised we have to write `copyRef` here. I thought we could do `RefPtr` = `Ref` without any function call.

Reply:
I will try. I think I assumed this wouldn't work without trying.

Reply:
It doesn't currently work:
loader/FrameLoader.cpp:1627:29: error: no viable overloaded '='
            requesterOrigin = requester->securityOrigin;

Reply:
I see now that we have the `RefPtr(Ref&&)` constructor but not the `RefPtr(const Ref&)` constructor. Not sure that’s really what we want in the future.

Reply:
It is indeed a bit annoying to have to copyRef() here. However, one benefit I guess is that it encourages developers to WTFMove() whenever possible.

Reply:
Yes, that was the original reasoning. But we don’t do that for copying a `RefPtr` to another `RefPtr` and we no longer do it for copying a `Ref` to a `Ref`, which we used to do, so it seems peculiar to still do that for `Ref` to `RefPtr`.